### ISSUE TITLE
diff is failing: utf-8 codec cant decode byte

### ISSUE THREAD

**Issue by beraldoleal** [avocado-framework/avocado#4177]
First job:
```bash
(avocado) [local@laptop avocado]$ avocado run /bin/false
JOB ID     : 3c409f6ebb9ad0af3b0e868d81b1fef974471390
JOB LOG    : /home/local/avocado/job-results/job-2020-09-02T09.06-3c409f6/job.log
 (1/1) /bin/false: FAIL: Exited with status: '1', stdout: '' stderr: '' (0.06 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/local/avocado/job-results/job-2020-09-02T09.06-3c409f6/results.html
JOB TIME   : 0.29 s
```

Second job:
```bash
(avocado) [local@laptop avocado]$ avocado run /bin/true
JOB ID     : db740a718c013d6322e8636eaec77cd1cbda6d03
JOB LOG    : /home/local/avocado/job-results/job-2020-09-02T09.06-db740a7/job.log
 (1/1) /bin/true: PASS (0.01 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/local/avocado/job-results/job-2020-09-02T09.06-db740a7/results.html
JOB TIME   : 0.29 s
```

diff:

```bash
(avocado) [local@laptop avocado]$ avocado diff 3c409f6ebb9ad0af3b0e868d81b1fef974471390 db740a718c013d6322e8636eaec77cd1cbda6d03
Avocado crashed unexpectedly: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
You can find details in /home/local/src/venvs/avocado/data/crashes/avocado-traceback-2020-09-02_09:06:50-z_g_kn0a.log
```

Log file content:

```Avocado crashed:
Traceback (most recent call last):
  File "/home/local/src/venvs/avocado/bin/avocado", line 11, in <module>
    load_entry_point('avocado-framework', 'console_scripts', 'avocado')()
  File "/home/local/src/beraldo/avocado/avocado/core/main.py", line 76, in main
    return app.run()
  File "/home/local/src/beraldo/avocado/avocado/core/app.py", line 112, in run
    return method(self.parser.config)
  File "/home/local/src/beraldo/avocado/avocado/plugins/diff.py", line 228, in run
    sysinfo_post1 = self._get_sysinfo(job1_dir, 'post')
  File "/home/local/src/beraldo/avocado/avocado/plugins/diff.py", line 418, in _get_sysinfo
    sysinfo.extend(sysinfo_file.readlines())
  File "/usr/lib64/python3.7/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte



```

**Comment by willianrampazzo** [comment]
It crashes when it tries to read `sysinfo/post/journalctl.gz` as a text file.

**Comment by clebergnu** [comment]
Fixed by #4224
